WebSocketNamespacedPodExecAsync: Support specifying command arguments

qmfrederik opened the pull request:
In the current implementation, `WebSocketNamespacedPodExecAsync` only allows a single command (`command` is a single string). 

The Kubernetes API allows you to pass a command with arguments, such as `/bin/bash -c "echo Hello, World!\nexit 0\n"`.

This PR changes the `command` argument to be an `IEnumerable<string>`, allowing you to pass parameters.

tg123 commented on src/Kubernetes.WebSocket.cs:
```diff
@@ -67,16 +72,24 @@ public Task<WebSocket> WebSocketNamespacedPodExecAsync(string name, string @name
-                { "stdin", stdin ? "1": "0"},
-                { "stdout", stdout ? "1": "0"},
-                { "tty", tty ? "1": "0"}
-            }).TrimStart('?');
```
that `trimstart ?` is because on net452, this api will NOT return with a `?` prefix (different from netstanard)

brendandburns replied:
Yeah, let's maintain this. (and is this covered by a test-case? if it's not it should be...)

tg123 commented on src/Kubernetes.WebSocket.cs:
```diff
@@ -67,16 +72,24 @@ public Task<WebSocket> WebSocketNamespacedPodExecAsync(string name, string @name
 
             uriBuilder.Path += $"api/v1/namespaces/{@namespace}/pods/{name}/exec";
 
+            List<string> queryParameters = new List<string>();
```
I assume you want multi commands param, but I suggest using `queryhelper`

queryhelper cloud help you add multiple params
```
    var q = QueryHelpers.AddQueryString(string.Empty, new Dictionary<string, string>
            {
                {"a", "b"}
            });

            q = QueryHelpers.AddQueryString(q, "command", "1");
            q = QueryHelpers.AddQueryString(q, "command", "1");
            q = QueryHelpers.AddQueryString(q, "command", "2");
            Console.WriteLine(q); // ?a=b&command=1&command=1&command=2
```

tg123 commented on src/Kubernetes.WebSocket.cs:
```diff
@@ -37,6 +37,11 @@ public Task<WebSocket> WebSocketNamespacedPodExecAsync(string name, string @name
                 throw new ArgumentNullException(nameof(command));
             }
 
+            if (command.Count() == 0)
```
suggest: use `Any()` instead of `Count()`

qmfrederik commented:
@tg123 Thanks for the prompt review! I made the changes requested, let me know if it looks OK now.

brendandburns commented on src/IKubernetes.WebSocket.cs:
```diff
@@ -52,7 +52,7 @@ public partial interface IKubernetes
         /// <return>
         /// A <see cref="ClientWebSocket"/> which can be used to communicate with the process running in the pod.
         /// </return>
-        Task<WebSocket> WebSocketNamespacedPodExecAsync(string name, string @namespace = "default", string command = "/bin/bash", string container = null, bool stderr = true, bool stdin = true, bool stdout = true, bool tty = true, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
Let's maintain this signature and add one that takes an `IEnumerable` I don't want to break the API contract for users of this library.

qmfrederik commented:
OK - I added overloads that take a `string` and `IEnumerable<string>`, so we keep the API stable. Also addressed the other feedback.

There's a unit test for `WebSocketNamespacedPodExecAsync`, not sure if you run unit tests on `net451`, though. (You can set up a CI build for `net451` using AppVeyor)

Let me know if you need anything else.

brendandburns commented:
Thanks!
